File: app/src/main/java/com/saizad/mvvmexample/components/MVVMExamplePageFragment.java
package com.saizad.mvvmexample.components;

import androidx.annotation.NonNull;

import com.saizad.mvvm.CurrentUser;
import com.saizad.mvvm.Environment;
import com.saizad.mvvm.SaizadLocation;
import com.saizad.mvvm.ViewModelProviderFactory;
import com.saizad.mvvm.pager.BasePage;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

abstract public class MVVMExamplePageFragment<VM extends MVVMExampleViewModel> extends BasePage<VM> {

    @Inject
    public Environment mainEnvironment;
    @Inject
    public ViewModelProviderFactory viewModelProviderFactory;
    @Inject
    SaizadLocation gpsLocation;

    @NonNull
    @Override
    public SaizadLocation appLocation() {
        return gpsLocation;
    }

    @NonNull
    @Override
    public ViewModelProviderFactory viewModelProviderFactory() {
        return viewModelProviderFactory;
    }

    @NotNull
    @Override
    public Environment environment() {
        return mainEnvironment;
    }

    protected CurrentUser currentUser() {
        return mainEnvironment.currentUser();
    }
}

File: app/src/main/java/com/saizad/mvvmexample/components/MVVMExamplePageFragment.kt
package com.saizad.mvvmexample.components

import com.saizad.mvvm.SaizadLocation
import com.saizad.mvvm.pager.BasePage
import com.saizad.mvvmexample.MVVMExampleCurrentUser
import sa.zad.easypermission.PermissionManager
import javax.inject.Inject

abstract class MVVMExamplePageFragment<VM : MVVMExampleViewModel> : BasePage<VM>() {
    
    @Inject
    lateinit var permissionManager: PermissionManager

    @Inject
    lateinit var gpsLocation: SaizadLocation

    @Inject
    lateinit var mvvmExampleCurrentUser: MVVMExampleCurrentUser
    
    override fun appLocation(): SaizadLocation {
        return gpsLocation
    }

    override fun permissionManager(): PermissionManager {
        return permissionManager
    }

}